### ISSUE TITLE
shader_prepass example panics

### ISSUE THREAD

**Issue by rparrett** [bevyengine/bevy#7879]
## Bevy version

main
bisected to #7784

## Relevant system information
```
AdapterInfo { name: "Apple M1 Max", vendor: 0, device: 0, device_type: IntegratedGpu, driver: "", driver_info: "", backend: Metal }
SystemInfo { os: "MacOS 13.1 ", kernel: "22.2.0", cpu: "Apple M1 Max", core_count: "10", memory: "64.0 GiB" }
```

## What you did

`cargo run --example shader_prepass`

## What went wrong

```
thread 'Compute Task Pool (1)' panicked at 'Resource requested by bevy_pbr::render::light::queue_shadows<shader_prepass::PrepassOutputMaterial> does not exist: bevy_pbr::prepass::PrepassPipeline<shader_prepass::PrepassOutputMaterial>', /Users/me/src/bevy/crates/bevy_ecs/src/system/system_param.rs:454:17
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
thread 'Compute Task Pool (2)' panicked at 'Resource requested by bevy_pbr::render::light::queue_shadow_view_bind_group<shader_prepass::PrepassOutputMaterial> does not exist: bevy_pbr::prepass::PrepassPipeline<shader_prepass::PrepassOutputMaterial>', /Users/me/src/bevy/crates/bevy_ecs/src/system/system_param.rs:454:17
thread 'Compute Task Pool (1)' panicked at 'A system has panicked so the executor cannot continue.: RecvError', crates/bevy_ecs/src/schedule/executor/multi_threaded.rs:194:60
thread '<unnamed>' panicked at 'called `Option::unwrap()` on a `None` value', /Users/me/src/bevy/crates/bevy_tasks/src/task_pool.rs:371:49
thread 'Compute Task Pool (1)' panicked at 'called `Result::unwrap()` on an `Err` value: RecvError', crates/bevy_render/src/pipelined_rendering.rs:136:45
thread 'main' panicked at 'called `Option::unwrap()` on a `None` value', /Users/me/src/bevy/crates/bevy_tasks/src/task_pool.rs:371:49
```

**Comment by geieredgar** [comment]
I believe this will be fixed by #7878. It is caused by this:
```rust
.add_plugin(MaterialPlugin::<PrepassOutputMaterial> {
            // This material only needs to read the prepass textures,
            // but the meshes using it should not contribute to the prepass render, so we can disable it.
            prepass_enabled: false,
            ..default()
        })
```
This disables the `PrepassPlugin<PrepassOutputMaterial>`, which is required for the shadow mapping to work.

**Comment by geieredgar** [comment]
I can confirm that the example does now work on my machine when using #7878. The dependency on the disabled `PrepassPlugin` was not the only issue, the custom material relied on `prepass.wgsl`, which did not always define a fragment shader entry point.
